File: JavaScript/Math-Array-String/48.js
// 2108. Find First Palindromic String in the Array

/**
 * @param {string[]} words
 * @return {string}
 */
var firstPalindrome = function(words) {
    for (let word of words) {
        if (word === word.split('').reverse().join('')) {
            return word; // Return the first palindromic string
        }
    }
    return ""; // Return an empty string if no palindrome is found
};

// Example usage:
let words1 = ["abc", "car", "ada", "racecar", "cool"];
console.log(firstPalindrome(words1)); // Output: "ada"

let words2 = ["notapalindrome", "racecar"];
console.log(firstPalindrome(words2)); // Output: "racecar"

File: Python/Math-Array-String/48.py
# 2108. Find First Palindromic String in the Array

from typing import List

class Solution:
    def firstPalindrome(self, words: List[str]) -> str:
        for word in words:
            if word == word[::-1]:  # Check if the word is the same forward and backward
                return word
        return ""  # Return an empty string if no palindrome is found
    

sol = Solution()
output = sol.firstPalindrome(["abc", "car", "ada", "racecar", "cool"])
print(output) # Output = ada